model.h

```diff
@@ -33,10 +45,12 @@ struct Model {
 
   DType weight_dtype() const { return _weight_dtype; }
 
+  void RestrictStates();
+
   // TODO:
   std::vector<Tensor> _embd_weights;
 
-private:
+public:
```

Comment:
I'm confused that I can't pass the compilation if keeping it private. Could you please help to take a look of it? @daquexian